src/zenml/model/model.py

```diff
@@ -878,21 +887,24 @@ def _prepare_model_version_before_step_launch(
         if self_copy.model_version_id is None:
             model_version_response = self_copy._get_or_create_model_version()
 
+            client = Client()
             # update the configured model version id in runs accordingly
             if step_run:
-                Client().zen_store.update_run_step(
+                client.zen_store.update_run_step(
                     step_run_id=step_run.id,
                     step_run_update=StepRunUpdate(
                         model_version_id=model_version_response.id
                     ),
                 )
+                step_run = client.zen_store.get_run_step(step_run.id)
```

Comment:
This is just to fetch the new response, correct? I think all our `update_...` methods in the zen store already return the updated model, so we can save this request (and same for the one below)

Reply:
Thanks, switched